WordPress/src/main/java/org/wordpress/android/ui/reader/viewmodels/ReaderPostDetailViewModel.kt

```diff
@@ -199,16 +229,54 @@ class ReaderPostDetailViewModel @Inject constructor(
         )
     }
 
+    private fun updateFollowButtonUiState(
+        currentUiState: ReaderPostDetailsUiState,
+        isFollowed: Boolean
+    ) {
+        val updatedFollowButtonUiState = currentUiState
+                .headerUiState
+                .followButtonUiState
+                .copy(isFollowed = isFollowed)
+
+        val updatedHeaderUiState = currentUiState
+                .headerUiState
+                .copy(followButtonUiState = updatedFollowButtonUiState)
+
+        _uiState.value = currentUiState.copy(headerUiState = updatedHeaderUiState)
+    }
+
+    private fun updatePostActions(post: ReaderPost) {
```

Comment:
Question (❓): Since I didn't dig too deeply into these changes, would you mind explaining it in a sentence? From the commit message I couldn't understand the full picture in terms of how it used be behave:

```
Refactor: Update only post actions to preserve related posts ui state on post action handle
```

Reply:
Same as above

I made this change because related posts are fetched separately and are not part of post model. Earlier when a post action (like/ bookmark etc) were performed `ReaderPostDetailsUiState` was recreated using the modified post model. Since modified post model doesn't contain related posts, I chose to update just the actions state and not recreating entire ui state.

Reply:
Thanks so much Ashita for explaining all that to me, much much appreciated! 🙏